### PR TITLE
Replace configuration loading with Viper

### PR DESCRIPTION
**Pull request by JoelSpeed**
## Description

This is the first in a series of PRs that I'm going to introduce to rework our configuration loading to use Viper and PFlag instead of our more legacy configuration.

This currently retains all existing behaviour of options loading but does the following:
- Uses Viper instead of go-options to merge options (with the same preference)
- Uses AutoEnv so we will be able to remove the env tags from the Options struct
- Tests the loading behaviour of our config (which we didn't really have before)
- Will eventually allow us to remove the slice flag code we use as PFlag has this built in
- Returns an error when loading if an options is provided in the config file that does not match the options struct (Fixes: #500)

## Motivation and Context

Code clean up/ make config easier (long term goal is for config to be structured)

## How Has This Been Tested?

Built and ran the code locally setting variables from config file, env and flags, existing behaviour is retained

Not yet tested, but added a test to check the environment will be loaded correctly

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
I need to verify that this doesn't cause any issues with flag merging since that's not in our codebase

**Comment by JoelSpeed** [comment]
I verified this does in fact work as expected and retains the original behaviour, I thought I had commented but it seems not 🤔

File: pkg/apis/options/load.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		typ = val.Elem().Type()
+	} else {
+		typ = val.Type()
+	}
+
+	for i := 0; i < typ.NumField(); i++ {
+		// pull out the struct tags:
+		//    flag - the name of the command line flag
+		//    cfg - the name of the config file option
+		//    env - the name of the environment variable
```
Comment (steakunderscore) [inline pkg/apis/options/load.go:68]: Looks like you are using automatic env instead

Comment (JoelSpeed) [reply pkg/apis/options/load.go:68]: Good catch with fix

**Comment by JoelSpeed** [comment]
I had left the tags in for now, with the intention of removing them in a follow up PR (#489)

**Comment by JoelSpeed** [comment]
One thought I've just had is that this will make flags require `--` rather than allowing either `-` or `--` as in the regular flag package, I'll need to add a note to the changelog that this is a breaking change

I don't think this is too bad of a change, not sure 🤔

**Comment by steakunderscore** [comment]
> One thought I've just had is that this will make flags require `--` rather than allowing either `-` or `--` as in the regular flag package, I'll need to add a note to the changelog that this is a breaking change
I'd be fine with this so long as a breaking change note is added.

**Comment by steakunderscore** [comment]
@JoelSpeed Ping me when you have updated this so we can get it merged

**Pull request merged**